core/net/mac/tsch/tsch-slot-operation.c

```diff
@@ -370,6 +371,24 @@ update_neighbor_state(struct tsch_neighbor *n, struct tsch_packet *p,
   return in_queue;
 }
 /*---------------------------------------------------------------------------*/
+static void
+tsch_radio_on(int inside_slot, int before_slot)
+{
+  if((!TSCH_RADIO_ON_DURING_TIMESLOT && inside_slot)
+      || (TSCH_RADIO_ON_DURING_TIMESLOT && before_slot)) {
+    NETSTACK_RADIO.on();
+  }
+}
+/*---------------------------------------------------------------------------*/
+static void
+tsch_radio_off(int inside_slot, int after_slot)
+{
+  if((!TSCH_RADIO_ON_DURING_TIMESLOT && inside_slot)
+      || (TSCH_RADIO_ON_DURING_TIMESLOT && after_slot)) {
+    NETSTACK_RADIO.off();
+  }
+}
```

Comment:
I find the two functions a bit hard to get, after all, if you're inside a slot how could you also be after?
How abut two parameters `inside_slot`, `force`?

Also, have you considered using the global `tsch_in_slot_operation`?